src/SocketSession.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. 
+
+#ifndef GROUP5_SRC_SOCKETSESSION_H_	
+#define GROUP5_SRC_SOCKETSESSION_H_
+
+#include <string>
+
+#include "SocketUtility.h"
+
+/** @brief Handle the upgrading http request*/
+class SocketSession {
+private:
+    /** @brief socket id.*/
+    int socket_;
+
+    /** @brief Receive stream package.
+     *  @param msg stream package's content.
+     */
+    void receiveMessage(const std::string &msg);
+    /** @brief Handle SocketMessage.
+     *  @param message Decoded SocketMessage.
+     */
+    void handleMessage(const SocketUtility::WSMessage &message);
+    /** @brief Pakage Message and Send.
+     *  @param message Decoded SocketMessage.
+     */
+    void sendMessage(const SocketUtility::WSMessage &message);
+
+public:
+    SocketSession(const int &_socket);
+    SocketSession(const SocketSession &) = delete;
+ 
+    /** @brief Run socketSession after initial.*/
+    void run();
+    /** @brief Close webSocket session.*/
+    void close();
+
+    SocketSession& operator=(const SocketSession&) = delete;
```

Comment:
Please group copy constructor and copy assignment operator together and add comment that this class is not copyable.

Reply:
Done